sdk/core/System.ClientModel/samples/ModelReaderWriter.md

```diff
@@ -53,4 +53,35 @@ string json = @"{
       ""z"": 3
     }";
 OutputModel? model = JsonSerializer.Deserialize<OutputModel>(json, options);
+```
+
+## Using a Proxy
+
+In more advanced scenarios a library user might want to override the behavior of how a model is read or written.
+In this case you can implement your own class which implements the same `IPersistableModel<T>` or `IJsonModel<T>` and register it with the `ModelReaderWriterOptions`.
+
+The example below shows how to write a persitable model using a proxy to `BinaryData`
+
+```C# Snippet:Readme_Read_Proxy
+InputModel model = new InputModel();
+
+ModelReaderWriterOptions options = new ModelReaderWriterOptions("W");
+options.AddProxy(typeof(InputModel), new InputModelProxy());
+
+BinaryData data = ModelReaderWriter.Write(model, options);
+```
+
+The example below shows how to read JSON to create a strongly-typed model instance using a proxy.
+
+```C# Snippet:Readme_Write_Proxy
+string json = @"{
+      ""x"": 1,
+      ""y"": 2,
+      ""z"": 3
+    }";
+
+ModelReaderWriterOptions options = new ModelReaderWriterOptions("W");
+options.AddProxy(typeof(OutputModel), new OutputModelProxy());
```

Comment:
Would it be possible to illustrate how the other public APIs being added are intended to be used as well?

Reply:
Discussed offline - the other APIs are used internally by proxy implementations, to read and write data in a custom way.